### ISSUE TITLE
update changelog: escape underscore characters

### ISSUE THREAD

**Issue by gabor** [grafana/grafana-github-actions#28]
we had pull-requests with `_` characters in the title, for example `handle $__interval`. the changelog-generator should markdown-escape the `_`, but it is not doing that currently.

**Comment by tolzhabayev** [comment]
@gabor can you elaborate a bit more on why it should escape it with a backslash as it looks like github can handle underscores pretty well even when unescaped?

**Comment by gabor** [comment]
hi @tolzhabayev , thanks for looking at this issue!

the CI jobs that validate the changelog-markdown-file will fail in some situations. for example, i made a test pull request where i removed the escaping, and see how it failed: https://github.com/grafana/grafana/pull/42829

basically, if you run `yarn run prettier:check`, it will complain if underscores are not escaped (seems not in every case, but it does for the case of `$__interval` and `$__interval_ms`)

**Comment by tolzhabayev** [comment]
@gabor I have looked more into it and it seems it should be [working](https://github.com/grafana/grafana-github-actions/blob/main/update-changelog/index.ts#L47-L57). I have tested it locally and it works for me 🤷 
I have created a [PR](https://github.com/grafana/grafana-github-actions/pull/44) to bump the `prettier` version, maybe that would help keeping it similar to the `grafana` repository. Otherwise it would be good to see which execution failed to escape (maybe there is an error message).

**Comment by gabor** [comment]
@tolzhabayev hmm, i think the prettier that is failing is in the `grafana/grafana` repo... you can reproduce it like this:
- use the `grafana/grafana` repo
- checkout the branch from my test-failed-PR `git checkout origin/gabor/changelog-escape-test`
- `yarn install`
- `git diff` .. this should show no changes
- `yarn run prettier:write`
- `git diff` .. this will show that prettier wants to change the `CHANGELOG.md` file

so, in short:
1. we have a github action that generates a changed `CHANGELOG.md`
2. the `prettier` check in `grafana/grafana` is not happy with the format of that `CHANGELOG.md`

so, if you want `prettier`-check in [2] be happy with the changes produces in [1], we either need to change [1], or change [2].

does this help?

maybe i'm misunderstanding something in the question, sorry 😄
